### PR TITLE
Implements multi-index support in Dataframe.filter

### PR DESCRIPTION
**Pull request by beobest2**
Implementing multi-index support in `DataFrame.filter`

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -9328,34 +9328,58 @@ def filter(self, items=None, like=None, regex=None, axis=None):
...
+                        else:
+                            col = col | (index_scols[0] == F.lit(item))
+                elif len(index_scols) > 1:
+                    # for multi-index
+                    cols = []
+                    col = None
+                    for item in items:
+                        if col is None:
+                            for i, element in enumerate(item):
+                                if col is None:
```
Comment (itholic) [inline databricks/koalas/frame.py:9345]: Should we check this here again? I think this is already checked at line 9343.

Comment (beobest2) [reply databricks/koalas/frame.py]: I've  just removed double check

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -9328,34 +9328,57 @@ def filter(self, items=None, like=None, regex=None, axis=None):
...
+                    cols = []
+                    col = None
+                    for item in items:
+                        for i, element in enumerate(item):
+                            if col is None:
+                                col = index_scols[i] == F.lit(element)
+                            else:
+                                col = col & (index_scols[i] == F.lit(element))
+                        cols.append(col)
+                        col = None
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py:9349]: I think you don't need to nullify the col here.

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -9328,34 +9328,57 @@ def filter(self, items=None, like=None, regex=None, axis=None):
...
+                    for c in cols:
+                        if col is None:
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py:9351]: Is it correct? Seems `col` is always going to be `None`.

File: databricks/koalas/frame.py [hunk 4]
```diff
@@ -9328,34 +9328,57 @@ def filter(self, items=None, like=None, regex=None, axis=None):
...
+                    for item in items:
+                        if col is None:
+                            col = index_scols[0] == F.lit(item)
+                        else:
+                            col = col | (index_scols[0] == F.lit(item))
+                elif len(index_scols) > 1:
+                    # for multi-index
+                    cols = []
+                    col = None
+                    for item in items:
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py:9342]: Let's check if `item` is a tuple and throw the same error like pandas's

Comment (HyukjinKwon) [inline databricks/koalas/frame.py:9342]: Let's do something like:

```python
col = None
for item in items:
    # Error if item is not a tuple
    # Error if item is empty
    midx_col = None
    for i, element in enumerate(item):
        if midx_col is None:
            midx_col = index_scols[i] == F.lit(element)
        else:
            midx_col = midx_col & (index_scols[i] == F.lit(element))
    if col is None:
        col = midx_col
    else:
        col = col | midx_col
```

File: databricks/koalas/frame.py (hunk 3 above)
Comment (beobest2) [reply databricks/koalas/frame.py:9352]: Now, it is checked only when the first value is entered in `col`.

**Pull request merged**